Hast.Abstractions/Hast.Vitis.Abstractions/Services/AzureHardwareImplementationComposerBuildProvider.cs

```diff
@@ -0,0 +1,189 @@
+                context =>
+                    File.Exists(
+                        UpdateBinaryPath(
+                            VitisHardwareImplementationComposerBuildProvider
+                                .GetBinaryPath(context.Configuration, context.HardwareDescription))));
+        }
+
+        private async Task PerformAttestationAsync(
+            AzureAttestationConfiguration configuration,
+            string binaryPath,
+            string sharedAccessSignature)
+        {
+            _logger.LogInformation("Sending attestation start request...");
+            var instanceId = (await GetResponseAsync<AzureResponseData>(
+                configuration.StartFunctionUrl,
+                new AzureStartPostData(configuration)
+                {
+                    BlobContainerSignature = sharedAccessSignature,
+                    Container = BlobContainerName,
+                    NetlistName = Path.GetFileName(binaryPath),
+                })).InstanceId;
+
+            _logger.LogInformation(
+                "Attestation request was submitted successfully with orchestration instance ID: {0}",
+                instanceId);
+            _logger.LogInformation("Checking the status of attestation using {0}", configuration.PollFunctionUrl);
+
+            while (true)
+            {
+                var (statusText, output) = await GetResponseAsync<AzurePollResponseData>(
+                    configuration.PollFunctionUrl,
+                    new AzurePollPostData(configuration, instanceId));
+                var statusUpper = statusText.ToUpperInvariant();
+
+                _logger.LogInformation("Polled status: {0}", statusText);
+
+                if (statusUpper == "PENDING" || statusUpper == "RUNNING")
+                {
+                    await Task.Delay(TimeSpan.FromSeconds(30));
+                    continue;
+                }
+
+                var outputList = output?.ToArray() ?? Array.Empty<string>();
+                if (outputList.Contains("Attestation process succeeded"))
+                {
+                    _logger.LogInformation("Attestation process has succeeded.");
+                    return;
+                }
+
+                var outputString = outputList.Any()
+                    ? string.Join("\n", outputList.Select(item => "- " + item.Trim()))
+                    : "- Unknown issue";
+                _logger.LogError("Attestation failed:\n{0}", outputString);
+
+                throw new InvalidOperationException("Attestation failed.");
+            }
+        }
+
+        private async Task<T> GetResponseAsync<T>(Uri url, object post)
+            where T : AzureResponseData
+        {
+            const string jsonContentType = "application/json";
+
+            var postJson = JsonConvert.SerializeObject(post);
+            var postContent = new StringContent(postJson, Encoding.UTF8, jsonContentType);
+
+            using var client = new HttpClient();

```

Comment:
Nem egészen így kellene használni a `HttpClient`-et, mert jó megtévesztő az API-ja: https://docs.microsoft.com/en-us/dotnet/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests De amúgy ott van az `IAzureAttestationApi` RestEase-zel is, akkor miért szükséges low-levelbe menni itt?

Reply:
A RestEase-es csak egy maradvány. Azzal nem működött és úgy látszik elfelejtettem letörölni. Az a baj vele, hogy configból jön az API endpoint, viszont ha megpróbálom változóban átadni akkor mindenképp url escape-eli.

Reply:
Újra implementáltam a RestEase-t from scratch és most jó. Lehet a múltkor valami mást rontottam el az új endpoint kevésbé hisztis.